src/core/infer_request.cc

```diff
@@ -284,14 +284,41 @@ InferenceRequest::CopyAsNull(const InferenceRequest& from)
     *new_input->MutableShape() = input.second.Shape();
     *new_input->MutableShapeWithBatchDim() = input.second.ShapeWithBatchDim();
 
+    std::shared_ptr<MutableMemory> new_data_str;
+
+    // Custom handling for null requests in TYPE_STRING. The null requests in
+    // type string have to follow a certain structure. We set the first 4 bytes
+    // in the Tensor to represent the number of total available bytes.
+    if (inference::DataType::TYPE_STRING == input.second.DType()) {
```

Comment:
It seems like this only works if a string input has shape [ -1 ]. What if the shape is [ 2, 2 ]. Then you need to initialize the data to be 4 empty strings. So 16 bytes of zeros.

My thoughts:
- In second pass above when you have a string tensor you calculate its data size as "num_elements * 4" because you are going to make each element an empty string so only need 4 bytes per element. Remember the max across all string inputs as max_string_byte_size. 
- When allocating 'data' use the max of max_byte_size and max_string_byte_size.
- If max_string_byte_size > 0 then zero that many bytes at the beginning of data so that it can be used for the string tensor data
- In the below when appending data for string tensor you must use "num_elements * 4" instead of TotalByteSize.

Reply:
@deadeyegoodwin I see. I missed the multi-dimensional case. I'll fix this.